src/components/duck.js

```diff
@@ -11,35 +10,24 @@ AFRAME.registerComponent("duck", {
   },
 
   init: function() {
-    this.physicsSystem = this.el.sceneEl.systems.physics;
-    this.hasBody = false;
-    this.position = new CANNON.Vec3();
-    this.force = new CANNON.Vec3(0, this.data.initialForce, 0);
-    this.initialScale = this.el.object3D.scale.x;
-    this.maxScale = this.data.maxScale * this.initialScale;
-  },
-
-  play: function() {
-    this.physicsSystem.addComponent(this);
-  },
+    NAF.utils.getNetworkedEntity(this.el).then(networkedEntity => {
+      this.networkedEntity = networkedEntity;
 
-  pause: function() {
-    this.physicsSystem.removeComponent(this);
+      this.initialScale = this.networkedEntity.object3D.scale.x;
+      this.maxScale = this.data.maxScale * this.initialScale;
+    });
   },
 
-  beforeStep: function() {
-    if (this.el.body && NAF.utils.isMine(this.el)) {
-      const currentScale = this.el.object3D.scale.x;
+  tick: function() {
+    if (this.networkedEntity && NAF.utils.isMine(this.networkedEntity)) {
```

Comment:
If this component is not under a networked entity, can we use `this.el` and skip checking `isMine`?

Reply:
yep